test/unit/spec/LoginRouter_spec.js

```diff
@@ -175,6 +174,17 @@ function (Okta, Q, Backbone, SharedUtil, CryptoUtil, CookieUtil, Logger, OktaAut
       expect(Logger.warn).toHaveBeenCalledWith('Field not defined in schema', arg1);
     }
 
+    var getAutoPushResponse = function (response, autoPushVal) {
+      var responseCopy = Util.deepCopy(response);
+      var factors = responseCopy['response']['_embedded']['factors'];
+      var factor = _.findWhere(factors, {factorType: 'push', provider: 'OKTA'});
```

Comment:
nit: Since only ID is needed, you can directly do something like: 
`var id = _.findWhere(factors, {factorType: 'push', provider: 'OKTA'}).id;`